sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/HiveTypeCoercion.scala

```diff
@@ -753,6 +753,9 @@ object HiveTypeCoercion {
         case (DateType, TimestampType) => Cast(e, TimestampType)
         case (TimestampType, DateType) => Cast(e, DateType)
 
+        // Implicit cast for functions like utc_timestamp
+        case (_: NumericType, TimestampType) => Cast(e, TimestampType)
```

Comment:
that's because hive support passing a numeric value as the first parameter of utc_timestamp, and act just like casting this numeric value to Timestamp first. I didn't check other functions that accepts timestamp, but I think they should also work in this way, will check soon.